platform/src/ToolsManager.js

```diff
@@ -43,6 +61,50 @@ class ToolManager {
         }
     }
 
+    /**
+     * Checks whether url_placeholder has a port attached to it and if so, it returns the port
+     * @return integer|null
+     */
+    getPort(url_placeholder) {
+        var url_placeholder_regexp = url_placeholder.match(new RegExp(/{{BASE-URL}}:*[0-9]*/));
+        if (
+            url_placeholder_regexp != null &&
+            url_placeholder.indexOf(':') > 0
+            ){
+            return url_placeholder_regexp[0].split(':')[1];
+        }
+
+        return null;
+    };
+
+    /**
+     * Communicates with the discovery service to fetch the relevant path by the service port.
+     * @returns string|null
+     */
+    fetchPathByPort(port) {
```

Comment:
Given that the discovery service only provides static data, why do we need to run it as a separate service at all? Couldn't the lookup simply happen in this function directly?

Reply:
Service discovery is now incorporated in education platform and runs as front-end code.